docs/develop/java/kafka.mdx

```diff
@@ -2,142 +2,79 @@
+Define the Kafka cluster that Restate needs to connect to in the [Restate configuration file](/operate/configuration#configuration-file):
 
 ```toml
 [[ingress.kafka-clusters]]
 name = "my-cluster"
 brokers = ["PLAINTEXT://broker:9092"]
 ```
 
-Check the [configuration documentation](/operate/configuration) for more details.
-
-## Connect an event handler with a topic
-
-To connect event handlers to a Kafka topic, you need to create a subscription using the [Admin API](/references/admin-api):
+Check the [configuration docs](/operate/configuration) for more details.
+</SubtleStep>
 
-```bash
-curl <RESTATE_ADMIN_URL>/subscriptions \
-  -H 'content-type: application/json' \
-  -d '{"source": "kafka://<CLUSTER_NAME>/<TOPIC_NAME>", "sink": "component://<COMPONENT_NAME>/<METHOD_NAME>"}'
-```
-
-For example:
+<SubtleStep stepLabel="3" title="Subscribe the event handler to the Kafka topic">
+Let Restate forward events from the Kafka topic to the event handler by creating a subscription using the [Admin API](/references/admin-api):
 
 ```bash
-curl localhost:9070/subscriptions \
-    -H 'content-type: application/json' \
-    -d '{"source": "kafka://my-cluster/my-topic", "sink": "component://MyService/Handle"}'
+curl localhost:9070/subscriptions -H 'content-type: application/json' \
+    -d '{
+           "source": "kafka://my-cluster/my-topic",
+           "sink": "component://MyService/Handle",
+           "options": {"auto.offset.reset": "earliest"}
+        }'
 ```
 
 Once you've created a subscription, Restate immediately starts consuming events from Kafka.
+The handler will be invoked for each event received from Kafka.
 
-### Additional subscription configuration
-
-You can add additional configuration parameters to the subscription through the `options` field, for example to start consuming the topic from the beginning:
-
-```bash
-curl <RESTATE_ADMIN_URL>/subscriptions \
-  -H 'content-type: application/json' \
-  -d '{"source": "kafka://<CLUSTER_NAME>/<TOPIC_NAME>", "sink": "component://<COMPONENT_NAME>/<METHOD_NAME>", "options": {"auto.offset.reset": "earliest"}}'
-```
-
-The `options` field accepts any configuration parameter from [librdkafka configuration](https://github.com/confluentinc/librdkafka/blob/master/CONFIGURATION.md).
-
-### Listing subscriptions
-
-List the current subscriptions via:
-
-```bash
-curl <RESTATE_ADMIN_URL>/subscriptions
-```
+The `options` field is optional and accepts any configuration parameter from [librdkafka configuration](https://github.com/confluentinc/librdkafka/blob/master/CONFIGURATION.md).
 
-For example:
+</SubtleStep>
 
-```bash
-curl localhost:9070/subscriptions
-```
+<Admonition type="info" title="Managing subscriptions">
+Have a look at the [registration docs](/operate/registration#kafka-subscriptions) for more commands to manage subscriptions.
+</Admonition>
 
-### Deleting a subscription
+<details className="grey-details">
+    <summary>Raw payload support</summary>
```

Comment:
Maybe the name for this should be

_Accept Kafka records with any value_

or something like that. raw payloads is rather an internal name we use for this feature, but probably has little meaning outside.

Reply:
Maybe raw event? The word value has a different meaning for me. More like the content itself instead of the type of the content. I'll merge this PR and take it along in the next one.